### PR TITLE
feature(converter) - refactor converter

### PR DESCRIPTION
**Pull request by giladlevy7**

### CODE REVIEW

File: src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java [hunk 1]
```diff
@@ -4,40 +4,60 @@
...
+    private static final Map<String, String> MONTH_MAP = Map.ofEntries(Map.entry("jan", "01"),
+            Map.entry("feb", "02"),
+            Map.entry("mar", "03"), Map.entry("apr", "04"), Map.entry("may", "05"),
+            Map.entry("jun", "06"),
+            Map.entry("jul", "07"), Map.entry("aug", "08"), Map.entry("sep", "09"),
+            Map.entry("oct", "10"),
             Map.entry("nov", "11"), Map.entry("dec", "12"));
     public static final int MILLIS_LENGTH = 13;
 
+    // columns of type "date" and "time" have special formats. All others are using the datetime format
```
Comment (eitamring) [inline src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java]:
```suggestion
    // Columns of type "date" and "time" have special formats. The rest are using the datetime format
```

File: src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java [hunk 2]
```diff
@@ -85,53 +107,102 @@ public void converterFor(RelationalColumn column, ConverterRegistration<SchemaBu
...
+                            stringValue, isTime);
+
+                String result;
+                try {
+                    result = GetDateTimeFormat(stringValue, column);
+                }
+                catch (Exception e){
+                    System.out.printf(
+                            "[TimestampConverter.converterFor] ERROR! Using regex for conversion. rawValue: %s, " +
+                                    "isTime: %s", stringValue, isTime);
```
Comment (eitamring) [inline src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java:131]: If it's like C++(printf) isn't isTime Bool?
If so shouldn't it be like so?
```suggestion
                            "[TimestampConverter.converterFor] ERROR! Using regex for conversion. rawValue: %s, " +
                                    "isTime: %b", stringValue, isTime);
```

File: src/test/java/oryanmoshe/kafka/connect/util/TimestampConverterTests.java [hunk 3]
```diff
@@ -101,9 +116,7 @@ void converterTest(final String columnType, final String format, final String in
         Object actualResult = mockRegistration._converter.convert(input);
         System.out.println(actualResult);
         if (actualResult == null) {
```
Comment (eitamring) [inline src/test/java/oryanmoshe/kafka/connect/util/TimestampConverterTests.java]:
```suggestion
        if (actualResult == null) {
```
Honestly don't know if its the Java way
But if we do 
if (actualResult != null)
we can skip the if else 
WDYT?

**Review by eitamring — APPROVED** [review]

File: src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java (hunk 1 above)
Comment (giladlevy7) [reply src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java]: u caught me

File: src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java (hunk 2 above)
Comment (giladlevy7) [reply src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java:131]: nope :) Java does "to string" to bools; AFAIK %s 
I might be wrong though

File: src/test/java/oryanmoshe/kafka/connect/util/TimestampConverterTests.java (hunk 3 above)
Comment (giladlevy7) [reply src/test/java/oryanmoshe/kafka/connect/util/TimestampConverterTests.java]: u right, fixed

File: src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java [hunk 4]
```diff
@@ -58,23 +78,25 @@ public void configure(Properties props) {
...
             System.out.printf(
-                    "[TimestampConverter.configure] Finished configuring formats. this.strDatetimeFormat: %s, this.strTimeFormat: %s%n",
+                    "[TimestampConverter.configure] Finished configuring formats." +
+                            " this.strDatetimeFormat: %s, this.strTimeFormat: %s%n",
                     this.strDatetimeFormat, this.strTimeFormat);
     }
 
     @Override
     public void converterFor(RelationalColumn column, ConverterRegistration<SchemaBuilder> registration) {
         if (this.debug)
```
Comment (IsraelAlafi) [inline src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java:88]: use brackets `if (this.debug) { ... }`

File: src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java [hunk 5]
```diff
@@ -85,53 +107,102 @@ public void converterFor(RelationalColumn column, ConverterRegistration<SchemaBu
...
                 }
 
-                Long millis = getMillis(rawValue.toString(), isTime);
-                if (millis == null)
-                    return rawValue.toString();
+                String stringValue = rawValue.toString();
 
-                Instant instant = Instant.ofEpochMilli(millis);
-                Date dateObject = Date.from(instant);
                 if (this.debug)
```
Comment (IsraelAlafi) [inline src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java:117]: use brcukets

File: src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java [hunk 6]
```diff
@@ -85,53 +107,102 @@ public void converterFor(RelationalColumn column, ConverterRegistration<SchemaBu
...
-            return null;
-
-        if (timestamp.contains(":") || timestamp.contains("-")) {
-            return milliFromDateString(timestamp);
+    private String GetDateTimeFormat(String rawValue, RelationalColumn column) {
+        TemporalAccessor result = null;
+        String output;
+
+        try {
+            result = source_format.parse(rawValue);
```
Comment (IsraelAlafi) [inline src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java:146]: Nice one 💯 
i thought `parse` can return `null` , but you can see here:  https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html
only get date-time or exception

File: src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java [hunk 7]
```diff
@@ -85,53 +107,102 @@ public void converterFor(RelationalColumn column, ConverterRegistration<SchemaBu
...
+                throw dtpx;
+            }
         }
+        SimpleDateFormat formatter = getFormatterPerColumnType(column);
+        DateTimeFormatter target_format = DateTimeFormatter.ofPattern(formatter.toPattern(), Locale.ENGLISH);
+        output = target_format.format(result);
 
-        int excessLength = timestamp.length() - MILLIS_LENGTH;
-        long longTimestamp = Long.parseLong(timestamp);
+        if (this.debug)
```
Comment (IsraelAlafi) [inline src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java:174]: bruckets

File: src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java [hunk 8]
```diff
@@ -85,53 +107,102 @@ public void converterFor(RelationalColumn column, ConverterRegistration<SchemaBu
...
-            return longTimestamp;
+    private String convertMillisToDateTimeString(RelationalColumn column, String rawValue, Long millis) {
+        if (millis == null)
+            return rawValue;
 
-        if (excessLength < 0)
-            return longTimestamp * 24 * 60 * 60 * 1000;
+        Instant instant = Instant.ofEpochMilli(millis);
+        Date dateObject = Date.from(instant);
+        if (this.debug)
```
Comment (IsraelAlafi) [inline src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java:187]: same here

**Review by IsraelAlafi — CHANGES_REQUESTED** [review]
Overall looks good.
Nice fallback 👯

File: src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java (hunk 6 above)
Comment (giladlevy7) [reply src/main/java/oryanmoshe/kafka/connect/util/TimestampConverter.java:146]: cool!

**Review by IsraelAlafi — APPROVED** [review]

**Review by Alonreznik — APPROVED** [review]

**Pull request merged**